### ISSUE TITLE
Learning is very bad

### ISSUE THREAD

**Issue by BeauGeogeo** [NVlabs/gbrl#18]
Hi,

I wanted to use GBRL for my PhD researches but I get very bad results, even after some time spent to tune the hyperparameters.

When I use DiscreteCritic in my DQN algorithm, paying attention to use predict_target when needed or using the model with 
`requires_grad=True` when appropriate, the loss almost always explode and even when it does not, the model does not seem to learn anything meaningful as evaluation is really bad.

Trying another approach using ParametricActor as in the example of the documentation, my results are again very bad, even when I tune the hyperparameters like n_bins, lr, etc. The loss either alternate and often reaches big values, and even when it has a quite big negative value like -138.0 the model does not seem to have learned anything.

By the way, I'm learning on a custom environment with feature vectors of size 28 (the input dim), those are my obs (richer representations obtained from raw environment obs to be more precise).

My custom implementations of RL algos like DQN or learners I use like sb3 work well but not GBRL and I can't figure it out why.

I wanted to use it because our features are quite explainable by design and with the SHAP values of GBRL that could be really great.

I can provide the code if needed (but it is quite big, at least I can provide the most relevant snippets). 

What could explain such a sensitive behaviour and difficulty to learn ?

Thanks a lot for your help,

Best,

Geoffrey.

**Comment by benja263** [comment]
I found a bug in the logic of how targets were predicted in both the continuous and discrete critics. I'll release an update in the upcoming days. It should improve performance.

**Comment by BeauGeogeo** [comment]
Hi @benja263,

Thank you so much for the fix, now it works really well ! The learning is not faster as other methods we use but it works.

I noticed that ParametricActor can totally converge on a wumpus 3x3 grid with 1 goal 1 wumpus and a safe setting (no wumpus around the agent when it begins), but DQN learner with the policy represented as a DiscreteCritic struggles. It learns things but not the optimal even after parameter tuning, but anyway it does not seem buggy because on small instances of all our benchmarcks (wumpus on 2x2 grid for instance) it converges, so thank you again !

I nonetheless needs to open another issue related to the copy method but this one can be closed i think.
